src/rules/prefer-to-have-value.js

```diff
@@ -0,0 +1,82 @@
+/**
+ * @fileoverview prefer toHaveAttribute over checking  getAttribute/hasAttribute
+ * @author Ben Monro
+ */
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+export const meta = {
+  docs: {
+    category: "jest-dom",
+    description: "prefer toHaveValue over checking  element.value",
+    url: "prefer-to-have-value",
+    recommended: false,
+  },
+  fixable: "code",
+  messages: {
+    "use-to-have-value": `Prefer .toBeInTheDocument() for asserting DOM node existence`,
+  },
+};
+const messageId = "use-to-have-value";
+
+export const create = (context) => ({
+  // expect(element.value).toBe('foo') / toEqual / toStrictEqual
+  [`CallExpression[callee.property.name=/to(Be|(Strict)?Equal)$/][callee.object.arguments.0.property.name=value][arguments.0][callee.object.callee.name=expect]`](
```

Comment:
I'm not sure the clause `[arguments.0]` is making a difference here. Or is it?

Reply:
```suggestion
  [`CallExpression[callee.property.name=/to(Be|(Strict)?Equal)$/][callee.object.arguments.0.property.name=value][callee.object.callee.name=expect]`](
```